### PR TITLE
JUP-22 Add Discord Auth Provider

### PR DESCRIPTION
**Pull request by danielsuhcompsci**
Adds discord provider, also adds test user for preview deployments

### CODE REVIEW

File: src/server/auth.ts [hunk 1]
```diff
@@ -75,10 +84,34 @@ export const authOptions: NextAuthOptions = {
...
+            } as PreviewUser;
+          },
+        })
+      : (GoogleProvider({
+          clientId: env.GOOGLE_CLIENT_ID,
+          clientSecret: env.GOOGLE_CLIENT_SECRET,
+        }),
+        DiscordProvider({
+          clientId: process.env.DISCORD_CLIENT_ID ?? '',
+          clientSecret: process.env.DISCORD_CLIENT_SECRET ?? '',
```
Comment (RubenOlano) [inline src/server/auth.ts:113]: Make sure to just use `env` since its the typesafe object

Comment (danielsuhcompsci) [reply src/server/auth.ts:113]: For some reason they can be undefined. Should I just use '!'?

File: src/pages/settings.tsx [hunk 2]
```diff
@@ -21,6 +21,8 @@ const Settings = () => {
         reset();
       }, 2000);
 
+    console.log(session);
+
```
Comment (RubenOlano) [inline src/pages/settings.tsx:25]: Remove the logs before ready for merge

File: src/server/auth.ts [hunk 3]
```diff
@@ -44,12 +53,34 @@ declare module 'next-auth' {
...
+        return session;
+      }
+
       let metadata = await db.query.userMetadata.findFirst({
         where: (metadata) => eq(metadata.id, user.id),
       });
 
+      console.log(user);
+      console.log(session);
+
```
Comment (RubenOlano) [inline src/server/auth.ts:83]: This too

**Review by RubenOlano — CHANGES_REQUESTED** [review]

**Comment by RubenOlano** [comment]
For this can you also add instructions to the README for creating your own environment variables for this?

**Review by RubenOlano — APPROVED** [review]

**Pull request merged**